### PR TITLE
gcc-arm-none-eabi 8-2019-q3-update (new formula)

### PR DESCRIPTION
**Pull request by ProZsolt**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
This was removed from cask (https://github.com/Homebrew/homebrew-cask/pull/56802)

### CODE REVIEW

File: Formula/gcc-arm-none-eabi.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+class GccArmNoneEabi < Formula
+  desc "GNU Arm Embedded Toolchain"
+  homepage "https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm"
+  url "https://developer.arm.com/-/media/Files/downloads/gnu-rm/8-2019q3/RC1.1/gcc-arm-none-eabi-8-2019-q3-update-src.tar.bz2"
+  version "8-2019-q3-update"
+  sha256 "e8a8ddfec47601f2d83f1d80c0600f198476f227102121c8d6a5a781d0c2eeef"
+
+  env :std
```
Comment (fxcoudert) [inline Formula/gcc-arm-none-eabi.rb]: This is not acceptable in Homebrew core.

File: Formula/gcc-arm-none-eabi.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  depends_on "wget" => :build
```
Comment (fxcoudert) [inline Formula/gcc-arm-none-eabi.rb]: Resources shouldn't be downloaded by the build script behind our back.

File: Formula/gcc-arm-none-eabi.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      # Task III-11, IV-8, V-0 and V-1 generates package which we don't need
+      s.gsub! /^echo Task \[III-11\].*?popd/m, ""
+      s.gsub! /^echo Task \[IV-8\].*?popd/m, ""
+      s.gsub! /^echo Task \[V-0\].*?popd/m, ""
+      s.gsub! /^echo Task \[V-1\].*?popd/m, ""
+    end
+
+    system "./install-sources.sh"
+    system "./build-prerequisites.sh"
+    system "./build-toolchain.sh", "--skip_steps=manual"
```
Comment (fxcoudert) [inline Formula/gcc-arm-none-eabi.rb]: Can this be made to build closer to `arm-linux-gnueabihf-binutils`?

File: Formula/gcc-arm-none-eabi.rb (hunk 1 above)
Comment (ProZsolt) [reply Formula/gcc-arm-none-eabi.rb]: Fixed

File: Formula/gcc-arm-none-eabi.rb (hunk 2 above)
Comment (ProZsolt) [reply Formula/gcc-arm-none-eabi.rb]: Hopefully fixed. I have to wait to find out if it builds with homebrews' versions of packages.

File: Formula/gcc-arm-none-eabi.rb (hunk 3 above)
Comment (ProZsolt) [reply Formula/gcc-arm-none-eabi.rb]: Sadly no. `build-toolchain.sh` is a 952 line monster. Just the happy path is ~450. I don't think we want to reimplement it for homebrew and maintain future compatibility.

**Comment by fxcoudert** [comment]
Hi @ProZsolt I'm sorry but the build process for this is simply not maintainable enough for us to accommodate in Homebrew core. You can ask for reinclusion in Homebrew casks given this cannot fit in core.

**Pull request closed without merging**